Include/object.h

```diff
@@ -839,6 +839,10 @@ PyAPI_FUNC(PyObject *) PyType_GetModuleByDef(PyTypeObject *, PyModuleDef *);
 PyAPI_FUNC(int) PyType_Freeze(PyTypeObject *type);
 #endif
 
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= _Py_PACK_VERSION(3, 15)
+PyAPI_FUNC(PyObject *) PyType_GetModuleByToken(PyTypeObject *, const void *);
```

Comment:
I would prefer to name parameters, it helps to understand the API:

```suggestion
PyAPI_FUNC(PyObject *) PyType_GetModuleByToken(PyTypeObject *type, const void *token);
```